### ISSUE TITLE
Question regarding loss calculation

### ISSUE THREAD

**Issue by lkreiskoether** [tensorflow/tcav#64]
Hi all,

while adapting the repo code to my own requirements I came across the line for the definition of the loss calculation:

    with graph.as_default():
      self.y_input = tf.placeholder(tf.int64, shape=[None])

      self.pred = tf.expand_dims(self.ends['prediction'][0], 0)
      self.loss = tf.reduce_mean(
          tf.nn.softmax_cross_entropy_with_logits(
              labels=tf.one_hot(
                  self.y_input,
                  self.ends['prediction'].get_shape().as_list()[1]),
              logits=self.pred))
    self._make_gradient_tensors()
As far as I understand we are here using the model predictions (class probabilities) as logits for tf.nn.softmax_cross_entropy_with_logits. Based on https://www.tensorflow.org/api_docs/python/tf/nn/softmax_cross_entropy_with_logits I now wonder why we are using class probabilities instead of logits for this argument because the docs say that this op expects unscaled logits since it performs a softmax on the logits internally.

Is there any specific reason for us using this op in this way?

Thanks in advance!

**Comment by lkreiskoether** [comment]
Hi again,

do we have any updates regarding this problem? In this context I thought that one possible fix for that could be to just change to a op for the softmax calculation which expects class probabilities as input. So something like this could work for that:

```
with graph.as_default():
  self.y_input = tf.placeholder(tf.int64, shape=[None])

  self.pred = tf.expand_dims(self.ends['prediction'][0], 0)
  self.loss = tf.reduce_mean(
      tf.keras.losses.categorical_crossentropy(
          y_true=tf.one_hot(
              self.y_input,
              self.ends['prediction'].get_shape().as_list()[1]),
          y_pred=self.pred))
self._make_gradient_tensors()
```
What do you think?

Thanks in advance!

**Comment by BeenKim** [comment]
Hello, 

There isn't much deep reasons for doing this logit vs class probability. Since the one is just normalized version of the other, I think either would get you pretty much the same answer. :) Please reopen if see further issues.
